pkg/mgmt/snapshot/snapshot.go

```diff
@@ -124,23 +130,30 @@ func (c *SnapController) updateSnap(oldObj, newObj interface{}) {
 
 	// update on Snapshot CR does not make sense unless it is a deletion candidate
 	if c.isDeletionCandidate(newSnap) {
-		klog.Infof("Got update event for Snap %s/%s@%s", newSnap.Spec.VolGroup, newSnap.Labels[lvm.LVMVolKey], newSnap.Name)
+		klog.Infof("Got update event for Snapshot %s/%s@%s", newSnap.Spec.VolGroup, newSnap.Labels[lvm.LVMVolKey], newSnap.Name)
 		c.enqueueSnap(newSnap)
 	}
 }
 
 // deleteSnap is the delete event handler for LVMSnapshot
 func (c *SnapController) deleteSnap(obj interface{}) {
-	snap, ok := obj.(*apis.LVMSnapshot)
+	snap, ok := c.getStructuredObject(obj)
 	if !ok {
-		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
-		if !ok {
-			runtime.HandleError(fmt.Errorf("Couldn't get object from tombstone %#v", obj))
-			return
-		}
-		snap, ok = tombstone.Obj.(*apis.LVMSnapshot)
-		if !ok {
-			runtime.HandleError(fmt.Errorf("Tombstone contained object that is not a lvmsnap %#v", obj))
+		unstructuredObj, ok := obj.(*unstructured.Unstructured)
+		if ok {
```

Comment:
Using if-else is not the recommended way...
Try using a pattern like so...
```go
unstructuredObj, ok := obj.(*unstructured.Unstructured)

if !ok {
        runtime.HandleError(fmt.Errorf("couldnt type assert obj: %#v to unstructured obj", obj))
	return
}

tombStone := cache.DeletedFinalStateUnknown{}
err := runtimenew.DefaultUnstructuredConverter.FromUnstructured(unstructuredObj.UnstructuredContent(), &tombStone)
if err != nil {
	runtime.HandleError(fmt.Errorf("couldn't get object from tombstone %#v", obj))
	return
}
snap, ok = tombStone.Obj.(*apis.LVMSnapshot)
if !ok {
	runtime.HandleError(fmt.Errorf("tombstone contained object that is not a lvmvolume %#v", obj))
	return
}
```

Reply:
Incorporated